### PR TITLE
Vacuum enhancements

### PR DESCRIPTION
**Pull request by mperezfuster**
Added references to the following VACUUM enhancements:

1. Avoid scanning pages unnecessarily. New VACUUM option `DISABLE_PAGE_SKIPPING`.
- This change adds a VACUUM option `DISABLE_PAGE_SKIPPING`.
-  The option `--disable-page-skipping` has also been added to the vacuumdb command.

2. Add the ability to skip VACUUM and ANALYZE operations on tables that cannot be locked immediately. New option `SKIP_LOCKED` for VACUUM and ANALYZE commands.
- This change adds a VACUUM and ANALYZE option `SKIP_LOCKED`
-  The option `--skip-locked` has also been added to the vacuumdb and analyzedb commands.

3. Allow VACUUM to skip index cleanup with VACUUM option `INDEX_CLEANUP`
	- This change adds a VACUUM command option `INDEX_CLEANUP` .
	- It also adds a Rel option `vacuum_index_cleanup` for pg_class. 

4. Avoid useless heap-truncation attempts during VACUUM
	There is no new content for this change, only Release Note content.

5. Allow VACUUM to avoid unnecessary index scans 
	- New GUC: vacuum_cleanup_index_scale_factor

Questions for writers/R&D:

- Any other pages that need update you think?

For change number 3:
 -  I cannot find anywhere to reference the addition of the reloption `vacuum_index_cleanup`. I presume we don't document these?
- I was not able to find any reference to the new option `INDEX_CLEANUP` for the wraparound command `vacuumdb`. Has this been included?

For change number 5:
- What are the set classification values for the GUC `vacuum_cleanup_index_scale_factor`? Ie master/local, session/system, reload/restart/reinit.

Review sites:
https://docs-staging.vmware.com/en/VMware-Tanzu-Greenplum/mireia-vacuum/greenplum-database/GUID-ref_guide-sql_commands-VACUUM.html
https://docs-staging.vmware.com/en/VMware-Tanzu-Greenplum/mireia-vacuum/greenplum-database/GUID-ref_guide-sql_commands-ANALYZE.html
https://docs-staging.vmware.com/en/VMware-Tanzu-Greenplum/mireia-vacuum/greenplum-database/GUID-utility_guide-ref-vacuumdb.html
https://docs-staging.vmware.com/en/VMware-Tanzu-Greenplum/mireia-vacuum/greenplum-database/GUID-utility_guide-ref-analyzedb.html
https://docs-staging.vmware.com/en/VMware-Tanzu-Greenplum/mireia-vacuum/greenplum-database/GUID-ref_guide-config_params-guc-list.html

### CODE REVIEW

File: gpdb-doc/markdown/ref_guide/sql_commands/VACUUM.html.md [hunk 1]
```diff
@@ -49,6 +49,15 @@ VERBOSE
...
+
+INDEX_CLEANUP
+:   Specifies that `VACUUM` should attempt to remove index entries pointing to dead tuples. This is normally the desired behavior and is the default unless you override it by setting `vacuum_index_cleanup` to `false` for the table you run `VACUUM` against. Setting this option to `false` may be useful when you need to make vacuum run as quickly as possible, for example, to avoid imminent transaction ID wraparound. However, you do not perform index cleanup regularly, performance may suffer, because as the table is modified, indexes accumulate dead tuples and the table itself accumulates dead line pointers that cannot be removed until index cleanup completes. This option has no effect for tables that do not have an index. If you use if the `FULL` option, it will ignore the `INDEX_CLEANUP` option.
```
Comment (VMWareJennyBeth) [inline gpdb-doc/markdown/ref_guide/sql_commands/VACUUM.html.md:59]: However, you do not perform => However, if you do not perform

**Review by VMWareJennyBeth — CHANGES_REQUESTED** [review]
Just one correction, otherwise perfect!

File: gpdb-doc/markdown/ref_guide/sql_commands/VACUUM.html.md (hunk 1 above)
Comment (dyozie) [inline gpdb-doc/markdown/ref_guide/sql_commands/VACUUM.html.md:59]: If you use if the `FULL` option -> If you use the `FULL` option

**Review by dyozie — CHANGES_REQUESTED** [review]

**Review by dyozie — APPROVED** [review]

File: gpdb-doc/markdown/ref_guide/config_params/guc-list.html.md [hunk 2]
```diff
@@ -2946,6 +2946,14 @@ Enables updating of the process title every time a new SQL command is received b
 |-----------|-------|-------------------|
 |Boolean|on|local, session, reload|
 
+## <a id="vacuum_cleanup_index_scale_factor"></a>vacuum_cleanup_index_scale_factor
```
Comment (lisakowen) [inline gpdb-doc/markdown/ref_guide/config_params/guc-list.html.md:2949]: do we want to include the other info about this guc on the https://www.postgresql.org/docs/12/runtime-config-client.html page?  or is it not applicable?

File: gpdb-doc/markdown/ref_guide/config_params/guc_category-list.html.md [hunk 3]
```diff
@@ -323,6 +323,7 @@ These configuration parameters set defaults that are used for client connections
 - [statement_timeout](guc-list.html#statement_timeout)
 - [temp_tablespaces](guc-list.html#temp_tablespaces)
 - [vacuum_freeze_min_age](guc-list.html#vacuum_freeze_min_age)
+- [vacuum_cleanup_index_scale_factor](guc-list.html#vacuum_cleanup_index_scale_factor)
```
Comment (lisakowen) [inline gpdb-doc/markdown/ref_guide/config_params/guc_category-list.html.md:326]: prob move this above vacuum_freeze_min_age if we are listing alphabetically.

**Review by lisakowen — APPROVED** [review]
looks good, a few minor comments.

**Review by zxuejing — APPROVED** [review]

**Pull request merged**